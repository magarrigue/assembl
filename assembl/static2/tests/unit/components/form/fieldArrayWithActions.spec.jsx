import React from 'react';
import renderer from 'react-test-renderer';
import { Field } from 'react-final-form';

import DummyForm from './dummyForm';
import FieldArrayWithActions from '../../../../js/app/components/form/fieldArrayWithActions';

describe('FieldArrayWithActions component', () => {
  const initialValues = {
    persons: [{ firstname: 'Rosalinda', lastname: 'Tillman' }, { firstname: 'Lavinia', lastname: 'Kunde' }]
  };

  it('should render an array of fields with delete, move up, move down and a add button', () => {
    const props = {
      name: 'persons',
      renderFields: ({ name }) => (
        <React.Fragment>
          <Field name={`${name}.firstname`} component="input" label="Firstname" />
          <Field name={`${name}.lastname`} component="input" label="Firstname" />
        </React.Fragment>
      ),
      titleMsgId: 'profile.panelTitle',
      tooltips: {
        addTooltip: () => <div>Add</div>,
        deleteTooltip: () => <div>Delete</div>
      },
      withSeparators: true
    };
    const component = renderer.create(
      <DummyForm initialValues={initialValues}>
        <FieldArrayWithActions {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an array of fields with delete, move up, move down and a add button without separators', () => {
    const props = {
      name: 'persons',
      renderFields: ({ name }) => (
        <React.Fragment>
          <Field name={`${name}.firstname`} component="input" label="Firstname" />
          <Field name={`${name}.lastname`} component="input" label="Firstname" />
        </React.Fragment>
      ),
      titleMsgId: 'profile.panelTitle',
      tooltips: {
        addTooltip: () => <div>Add</div>,
        deleteTooltip: () => <div>Delete</div>
      },
      withSeparators: false
    };
    const component = renderer.create(
      <DummyForm initialValues={initialValues}>
        <FieldArrayWithActions {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});