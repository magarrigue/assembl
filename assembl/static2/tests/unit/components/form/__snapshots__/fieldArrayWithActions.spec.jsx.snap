// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldArrayWithActions component should render an array of fields with delete, move up, move down and a add button 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <div
      className="form-container"
    >
      <div
        className="title left"
      >
        <span
          className={undefined}
          style={undefined}
        >
          My account
        </span>
      </div>
      <div
        className="pointer right"
      >
        <div
          className="inline"
        >
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-down-bold grey"
            />
          </button>
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-delete grey"
            />
          </button>
        </div>
      </div>
      <div
        className="clear"
      />
      <input
        label="Firstname"
        name="persons[0].firstname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Rosalinda"
      />
      <input
        label="Firstname"
        name="persons[0].lastname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Tillman"
      />
      <div
        className="separator"
      />
    </div>
    <div
      className="form-container"
    >
      <div
        className="title left"
      >
        <span
          className={undefined}
          style={undefined}
        >
          My account
        </span>
      </div>
      <div
        className="pointer right"
      >
        <div
          className="inline"
        >
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-up-bold grey"
            />
          </button>
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-delete grey"
            />
          </button>
        </div>
      </div>
      <div
        className="clear"
      />
      <input
        label="Firstname"
        name="persons[1].firstname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Lavinia"
      />
      <input
        label="Firstname"
        name="persons[1].lastname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Kunde"
      />
      <div
        className="separator"
      />
    </div>
    <div
      className="plus margin-l"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      +
    </div>
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`FieldArrayWithActions component should render an array of fields with delete, move up, move down and a add button without separators 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <div
      className="form-container"
    >
      <div
        className="title left"
      >
        <span
          className={undefined}
          style={undefined}
        >
          My account
        </span>
      </div>
      <div
        className="pointer right"
      >
        <div
          className="inline"
        >
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-down-bold grey"
            />
          </button>
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-delete grey"
            />
          </button>
        </div>
      </div>
      <div
        className="clear"
      />
      <input
        label="Firstname"
        name="persons[0].firstname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Rosalinda"
      />
      <input
        label="Firstname"
        name="persons[0].lastname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Tillman"
      />
    </div>
    <div
      className="form-container"
    >
      <div
        className="title left"
      >
        <span
          className={undefined}
          style={undefined}
        >
          My account
        </span>
      </div>
      <div
        className="pointer right"
      >
        <div
          className="inline"
        >
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-up-bold grey"
            />
          </button>
          <button
            className="admin-icons btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <span
              className="assembl-icon-delete grey"
            />
          </button>
        </div>
      </div>
      <div
        className="clear"
      />
      <input
        label="Firstname"
        name="persons[1].firstname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Lavinia"
      />
      <input
        label="Firstname"
        name="persons[1].lastname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Kunde"
      />
    </div>
    <div
      className="plus margin-l"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      +
    </div>
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
